src/linear_algebra/multilinear.lean

```diff
@@ -1030,27 +1044,27 @@ variables {ι' : Type*} [decidable_eq ι'] [decidable_eq (ι ⊕ ι')] {R M₂}
 
 /-- A multilinear map on `Π i : ι ⊕ ι', M'` defines a multilinear map on `Π i : ι, M'`
 taking values in the space of multilinear maps on `Π i : ι', M'`. -/
-def curry_sum (f : multilinear_map R (λ x : ι ⊕ ι', M') M₂) :
-  multilinear_map R (λ x : ι, M') (multilinear_map R (λ x : ι', M') M₂) :=
-{ to_fun := λ u,
-  { to_fun := λ v, f (sum.elim u v),
+def curry_sum (f : (ι ⊕ ι' → M') →ₗₘ[R] M₂) :
+  (ι → M') →ₗₘ[R] ((ι' → M') →ₗₘ[R] M₂) :=
+{ to_fun := λ (u : ι → M'),
+  { to_fun := λ (v : ι' → M'), f (sum.elim u v),
```

Comment:
These casts are unfortunately now required here to eliminate the `typeinfo` nonsense.